Change de/compression code, add tests

dankwart-de opened the pull request:
Remove the working directory from all de/compression methods. Was not
used effectively.

Add tests for de/compression methods.

Leave some deprecated methods to keep compatibility.

vinjana committed:
Escaping bash variables.

Commit:
Change de/compression code, add tests

Remove the working directory from all de/compression methods. Was not
used effectively.

Add tests for de/compression methods.

Leave some deprecated methods to keep compatibility.

vinjana commented on src/main/groovy/de/dkfz/roddy/tools/RoddyIOHelperMethods.groovy:
```diff
@@ -28,148 +28,141 @@ import java.nio.file.attribute.PosixFilePermissions
-//            GString gString = "${wdPath} [[ -e ${_to} ]] && zip -dr9 ${zipTarget} || zip -r9 ${zipTarget}";
-            GString gString = "[[ -f \"${_to}\" ]] && rm ${_to}; ${wdPath} zip -r9 ${zipTarget} > /dev/null && md5sum ${_to}";
-            return gString;
+            GString gString = "[[ -f \"${_to}\" ]] && rm ${_to}; cd ${from.parent} && zip -r9 ${zipTarget} > /dev/null && md5sum ${_to}"
```
Then it should also be `rm "{_to}"`. With rm it would actually be particularly important, rigth?

```groovy
def _to = "/ upsSomeSpace"
```

vinjana replied:
Generally, we should think about using single quotes filenames. Even better, use the (upcoming) Bash-escaping function/service.

dankwart-de replied:
sure. I'll try to utilize the BashEscape stuff.

vinjana commented on src/main/groovy/de/dkfz/roddy/tools/RoddyIOHelperMethods.groovy:
```diff
@@ -214,57 +207,67 @@ class RoddyIOHelperMethods {
+    /**
+     *
+     * @param file
+     * @param targetBaseFolder The target folder to which the archives content will be extracted.
```
remove the "Base"

dankwart-de replied:
ok

vinjana commented on src/test/groovy/de/dkfz/roddy/tools/RoddyIOHelperMethodsSpecification.groovy:
```diff
@@ -0,0 +1,131 @@
+
+import static de.dkfz.roddy.tools.RoddyIOHelperMethods.NativeLinuxZipCompressor
+
+class RoddyIOHelperMethodsSpecification extends Specification {
```
It is conventional to call these "...Spec".

dankwart-de replied:
I'll rename them.

vinjana commented on src/main/groovy/de/dkfz/roddy/tools/RoddyIOHelperMethods.groovy:
```diff
@@ -28,148 +28,141 @@ import java.nio.file.attribute.PosixFilePermissions
      * Also produces a file named [zip.gz].md5
      */
-    public static class NativeLinuxZipCompressor extends Compressor {
+    static class NativeLinuxZipCompressor extends Compressor {
```
Why "Native"? Why "Linux? Why not "Bash"? Or "Posix"? What about `-d` and `-f` and the `if` syntax? Why make this a nested class? Should this not just all go into a "compressor" package? 

Sorry for all the questions.

dankwart-de replied:
No sorry necessary. Why NativeLinuxZip? Well it is written specifically for Linux utilizing zip/unzip and md5sum. Bash is not necessary. We could call it LinuxZipCompressor or so. Of course I can extract the classes.

vinjana commented on src/test/groovy/de/dkfz/roddy/tools/RoddyIOHelperMethodsSpecification.groovy:
```diff
@@ -0,0 +1,131 @@
+
+        def rangeOfFiles = ["a", "b", "c"]
+        File tempFolder = File.createTempDir()
+        tempFolder.deleteOnExit()
```
Did you check, whether this is deleted even if files are in there? We had this problem with tons of file before and I am not 100 sure where they came from. There is also a TemporaryFolder @Rule in junit4.

dankwart-de replied:
Nope did not. But I changed to a @Rule with TemporaryFolder

vinjana reviewed (commented):
Nice tests.

Commit:
Merge branch 'bash-services' into ChangeZipCompressionDecompressionCode

Commit:
Changes according to #22

Moved the compression classes to a compression package.

Moved the compression tests to a compression test package.

Added some more lines to the compression / decompression test cases.

Remove one obsolete test for decompression strings.

Use the String Bash Escaping service to properly escape filenames for
compression and decompression.

dankwart-de commented:
Discussed with vinjana, will merge then.
